tune_rf_with_cv.py

```
@@ -0,0 +1,318 @@
+    Tune Random Forest using cross-validation to avoid overfitting.
+    """
+
+    # Load data
+    print("Loading and preprocessing data...")
+    # X_train, X_test, y_train, y_test, sw_train, sw_test, groups_train = load_preprocessed_data()
+
+    # For demonstration, let's assume you have 20 features (12 PCs + 8 engineered)
+    n_features = 20
+
+    print(f"\nTotal features: {n_features}")
+    print(f"  - 12 PCA components from traces")
```

Ruff (0.14.4)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)